### PR TITLE
Fix incorrect log messages

### LINKED ISSUE #31: Document library dependencies
Hyperfoil may need some native dependencies, e.g. for ALPN support which requires OpenSSL (at least on older Java versions). Document that somewhere.

Normally `tcnative-boringssl` requires `openssl` and `apr`; on Fedora 30, though `libxcrypt-compat` is required as well.

### LINKED ISSUE #407: Non-constant string concatenation as argument to 'error()' logging call
There are some places where the log message is not properly computed, e.g.,

```java
log.error("Cannot start " + name, e);
```
And the concatenation is calculated beforehand the condition check. So if that logging message is in the hot path and gets called many times, e.g., 10K times but the condition check is always false, you will be concatenating 10K times with no reason.

Therefore, it should be refactored as:
```java
log.error("Cannot start {}", name, e);
```

### PR DESCRIPTION
**Pull request by mini0192**
## Fix incorrect log messages

Fixes #407

## Check List (check all the applicable boxes) 

- [ x ] My change requires changes to the documentation.
- [ x ] I have updated the documentation accordingly.
- [ x ] All new and existing tests passed.

> Note that the documentation is located at [github.com/Hyperfoil/hyperfoil.github.io](https://github.com/Hyperfoil/hyperfoil.github.io/) 

<details>
<summary>
How to backport a pull request to the current <em>stable</em> branch?
</summary>

In order to automatically create a **backporting pull request** please add `backport` label to the current pull request.

Then, as soon as the pull request is merged into the `master` branch, the process will try to automatically open a backporting pull request against the _stable_ branch. If something goes wrong, a comment will be added in the original pull request such that all people involved get notified.

> The `backport` label can be also added after the pull  request has been merged.

> If the process fails due to temporary problems, such as connectivity problems, consider removing and re-adding the `backport` label.
</details>

### CODE REVIEW

File: clustering/src/main/java/io/hyperfoil/clustering/ControllerVerticle.java [hunk 1]
```diff
@@ -1025,10 +1025,10 @@ public void listSessions(Run run, boolean includeInactive, BiConsumer<AgentInfo,
          Handler<AsyncResult<Void>> completionHandler) {
       invokeOnAgents(run, AgentControlMessage.Command.LIST_SESSIONS, includeInactive, completionHandler, (agent, result) -> {
          if (result.failed()) {
-            log.error("Agent " + agent + " failed listing sessions", result.cause());
+            log.error("Agent {} failed listing sessions", agent, result.cause());
             sessionStateHandler.accept(agent, "");
          } else if (result.result() instanceof Throwable) {
-            log.error("Agent " + agent + " has thrown an error while listing sessions", (Throwable) result.result());
+            log.error("Agent {}} has thrown an error while listing sessions", agent, (Throwable) result.result());
```
Comment (lampajr) [inline clustering/src/main/java/io/hyperfoil/clustering/ControllerVerticle.java:1031]:
```suggestion
            log.error("Agent {} has thrown an error while listing sessions", agent, (Throwable) result.result());
```

**Review by lampajr — CHANGES_REQUESTED** [review]
Hi @mini0192,

Looks awesome, thanks a lot for taking care of this issue! :heart: 

I left just one comment as I spot a double `}` that should be removed.

Could you please change the PR title to something a bit more meaningful, e.g., using the same commit description like `Fix incorrect log messages`?

File: clustering/src/main/java/io/hyperfoil/clustering/ControllerVerticle.java (hunk 1 above)
Comment (mini0192) [reply clustering/src/main/java/io/hyperfoil/clustering/ControllerVerticle.java:1031]: Thank you for answer
I will quickly fix it and PR it for you!

**Review by franz1981 — APPROVED** [review]
Nice one :heart: 
many thanks and welcome, that's your very first contribution here!

**Comment by mini0192** [comment]
> Nice one ❤️ many thanks and welcome, that's your very first contribution here!

Thank you. If there is anything I can help you with next time, I will leave a message!

**Review by lampajr — APPROVED** [review]
LGTM!

**Comment by lampajr** [comment]
Thanks a lot for this contribution @mini0192 :heart: !

**Pull request merged**